forgerock-authenticator/src/main/java/org/forgerock/android/auth/DefaultStorageClient.java

```diff
@@ -0,0 +1,236 @@
+ */
+public class DefaultStorageClient implements StorageClient {
+
+    //Alias to store keys
+    private static final String ORG_FORGEROCK_SHARED_PREFERENCES_KEYS = "org.forgerock.android.authenticator.KEYS";
+
+    //Settings to store the data
+    private static final String ORG_FORGEROCK_SHARED_PREFERENCES_DATA_ACCOUNT = "org.forgerock.android.authenticator.DATA.ACCOUNT";
+    private static final String ORG_FORGEROCK_SHARED_PREFERENCES_DATA_MECHANISM = "org.forgerock.android.authenticator.DATA.MECHANISM";
+    private static final String ORG_FORGEROCK_SHARED_PREFERENCES_DATA_NOTIFICATIONS = "org.forgerock.android.authenticator.DATA.NOTIFICATIONS";
+
+    //The SharedPreferences to store the data
+    private SharedPreferences accountData;
+    private SharedPreferences mechanismData;
+    private SharedPreferences notificationData;
+
+    private static final String TAG = DefaultStorageClient.class.getSimpleName();
+
+    public DefaultStorageClient(Context context) {
+        this.accountData = new SecuredSharedPreferences(context,
+                ORG_FORGEROCK_SHARED_PREFERENCES_DATA_ACCOUNT, ORG_FORGEROCK_SHARED_PREFERENCES_KEYS);
+        this.mechanismData = new SecuredSharedPreferences(context,
+                ORG_FORGEROCK_SHARED_PREFERENCES_DATA_MECHANISM, ORG_FORGEROCK_SHARED_PREFERENCES_KEYS);
+        this.notificationData = new SecuredSharedPreferences(context,
+                ORG_FORGEROCK_SHARED_PREFERENCES_DATA_NOTIFICATIONS, ORG_FORGEROCK_SHARED_PREFERENCES_KEYS);
+
+//        this.accountData = context.getApplicationContext()
+//                .getSharedPreferences(ORG_FORGEROCK_SHARED_PREFERENCES_DATA_ACCOUNT, Context.MODE_PRIVATE);
+//        this.mechanismData = context.getApplicationContext()
+//                .getSharedPreferences(ORG_FORGEROCK_SHARED_PREFERENCES_DATA_MECHANISM, Context.MODE_PRIVATE);
+//        this.notificationData = context.getApplicationContext()
+//                .getSharedPreferences(ORG_FORGEROCK_SHARED_PREFERENCES_DATA_NOTIFICATIONS, Context.MODE_PRIVATE);
+    }
+
+    @Override
+    public Account getAccount(String accountId) {
+        Gson gson = new Gson();
+        String json = accountData.getString(accountId, "");
+        return gson.fromJson(json, Account.class);
+    }
+
+    @Override
+    public List<Account> getAllAccounts() {
+        List<Account> accountList = new ArrayList<>();
+        Gson gson = new Gson();
+
+        Map<String,?> keys = accountData.getAll();
+        for(Map.Entry<String,?> entry : keys.entrySet()){
+            Logger.debug(TAG, "Account map values: ",entry.getKey() + ": " + entry.getValue().toString());
+            Account account = gson.fromJson(entry.getValue().toString(), Account.class);
+            if(account != null)
+                accountList.add(account);
+        }
+
+        return accountList;
+    }
+
+    @Override
+    public boolean removeAccount(Account account) {
```

Comment:
If remove account by id, the input can be just the account id instead of the Account object.

Reply:
It was previously implemented that way. I changed it to align with iOS implementation. I am taking a note to discuss it later.